doc/reference/dune-project/package.rst

```diff
@@ -26,6 +26,29 @@ is used when generating OPAM files (see :doc:`generate_opam_files`).
 
       Package dependencies, as :token:`~pkg-dep:dep_specification`.
 
+   .. describe:: (documentation <doc-spec>)
+
+      The documentation field allows to specify documentation options for the
+      package.
```

Comment:
I think we should also document that `(documentation "string")` is the same as `(documentation (url "string"))` for consistency and backward compatibility.